### PR TITLE
Resolve rotation crash

### PR DESCRIPTION
**Pull request by 276746**
# Bug:
The application crashed when the screen is rotated. This was due to the MainActivity recreating every dependency each time it was recreated. I added factories to each viewmodel to create them properly

The source of this bug was originally found by @ethangraham2001 who proposed a first quickfix, see closed PR#318.

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/ui/signin/LoginScreen.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/signin/LoginScreen.kt:1]: great job noticing that

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ItineraryViewModel.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ItineraryViewModel.kt:1]: I don't quite understand the @supress, but I'll get back to you on that in order to understand better

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModel.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ProfileViewModel.kt:1]: the comments and the functionning look good

**Review by imaneb942 — APPROVED** [review]
Everything looks good to me, great job to you and Ethan for this PR

File: app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ItineraryViewModel.kt
Comment (276746) [reply app/src/main/java/com/github/wanderwise_inc/app/viewmodel/ItineraryViewModel.kt:1]: It is just to suppress some type casting warning

**Pull request merged**